### PR TITLE
Support nuspec file check

### LINKED ISSUE #31: Check, that nuspec and packages.config has the same versions
For case when solution has nuget packages on result, nuspec file should have the same dependencies with packages.config files.

### PR DESCRIPTION
**Pull request by imanushin**
Fix #31

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -169,6 +169,7 @@ _UpgradeReport_Files/
 Backup*/
 UpgradeLog*.XML
 UpgradeLog*.htm
+.vs/
```
Comment (imanushin) [inline .gitignore]: msvs 2015, msvs 2017

File: src/SolutionCop.CommandLine/SolutionCop.CommandLine.csproj [hunk 2]
```diff
@@ -9,7 +9,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>SolutionCop.CommandLine</RootNamespace>
     <AssemblyName>SolutionCop</AssemblyName>
-    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
```
Comment (imanushin) [inline src/SolutionCop.CommandLine/SolutionCop.CommandLine.csproj]: MSVS 2017 does not support v4.0 by default. For current moment we can increase framework.
I'll use new features in next pull requests

File: src/SolutionCop.Core.Tests/SolutionCop.Core.Tests.csproj [hunk 3]
```diff
@@ -38,24 +38,24 @@
   <ItemGroup>
     <Reference Include="ApprovalTests, Version=3.0.0.0, Culture=neutral, PublicKeyToken=11bd7d124fc62e0f, processorArchitecture=MSIL">
       <HintPath>..\packages\ApprovalTests.3.0.11\lib\net40\ApprovalTests.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="ApprovalUtilities, Version=3.0.0.0, Culture=neutral, PublicKeyToken=11bd7d124fc62e0f, processorArchitecture=MSIL">
-      <HintPath>..\packages\ApprovalUtilities.3.0.11\lib\net35\ApprovalUtilities.dll</HintPath>
-      <Private>True</Private>
+      <HintPath>..\packages\ApprovalUtilities.3.0.11\lib\net45\ApprovalUtilities.dll</HintPath>
```
Comment (imanushin) [inline src/SolutionCop.Core.Tests/SolutionCop.Core.Tests.csproj]: Update-Package -Reinstall did this

**Comment by imanushin** [comment]
image

File: src/SolutionCop.DefaultRules/NuGet/NuspecDependencyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+﻿namespace SolutionCop.DefaultRules.NuGet
```
Comment (imanushin) [inline src/SolutionCop.DefaultRules/NuGet/NuspecDependencyInfo.cs]: I'll reuse this file from another rules which parse packages.config

File: src/SolutionCop.DefaultRules/ProjectRule.cs [hunk 5]
```diff
@@ -84,37 +84,8 @@ public ValidationResult ValidateAllProjects(XElement xmlRuleConfigs, params stri
 
         protected void ValidateConfigSectionForAllowedElements(XElement xmlElement, List<string> errors, params string[] allowedElementNames)
```
Comment (imanushin) [inline src/SolutionCop.DefaultRules/ProjectRule.cs]: Moved to static class, I reused this method

File: deployment/SolutionCop.nuspec [hunk 6]
```diff
@@ -6,7 +6,7 @@
     <title>SolutionCop</title>
     <authors>Andrey Lipatkin</authors>
     <owners>litee</owners>
-    <copyright>2015 Andrey Lipatkin</copyright>
+    <copyright>2017 Andrey Lipatkin</copyright>
```
Comment (Litee) [inline deployment/SolutionCop.nuspec]: Minor: 2015-2017

File: src/SolutionCop.DefaultRules.Tests/NuGet/NuspecHasTheSameVersionsWithPackagesConfigTest.cs [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+        public void ShouldUnionVersionsFromDifferentProjects()
+        {
+            var testSubfolder = "MultiplePackages";
+            var testFolder = TestDirectory.GetDirectories(testSubfolder).Single();
+
+            var xmlConfig = XElement.Parse(@"
+<NuspecHasTheSameVersionsWithPackagesConfig enabled='true'>
+  <ExcludePackagesOfProject projectName='IgnoredProject.csproj'/>  
+  <ExcludePackageId packageId='incorrect-ignored-global-package'/>
+  <Nupspec>
```
Comment (Litee) [inline src/SolutionCop.DefaultRules.Tests/NuGet/NuspecHasTheSameVersionsWithPackagesConfigTest.cs]: Typo

File: src/SolutionCop.DefaultRules.Tests/NuGet/NuspecHasTheSameVersionsWithPackagesConfigTest.cs [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+
+        [Fact]
+        public void ShouldUnionVersionsFromDifferentProjects()
+        {
+            var testSubfolder = "MultiplePackages";
+            var testFolder = TestDirectory.GetDirectories(testSubfolder).Single();
+
+            var xmlConfig = XElement.Parse(@"
+<NuspecHasTheSameVersionsWithPackagesConfig enabled='true'>
+  <ExcludePackagesOfProject projectName='IgnoredProject.csproj'/>
```
Comment (Litee) [inline src/SolutionCop.DefaultRules.Tests/NuGet/NuspecHasTheSameVersionsWithPackagesConfigTest.cs]: As discussed - please use standard <Exception>*</Exception> convention.

File: src/SolutionCop.DefaultRules.Tests/NuGet/NuspecHasTheSameVersionsWithPackagesConfigTest.cs [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+
+            ShouldPassNormally(xmlConfig, projects);
+        }
+
+        [Fact]
+        public void ShouldFailForInvalidNuspecFilesLookupPatterns()
+        {
+            var xmlConfig = XElement.Parse(@"
+<NuspecHasTheSameVersionsWithPackagesConfig enabled='true'>
+  <Nupspec>
```
Comment (Litee) [inline src/SolutionCop.DefaultRules.Tests/NuGet/NuspecHasTheSameVersionsWithPackagesConfigTest.cs]: Multiple typos

File: deployment/SolutionCop.nuspec (hunk 6 above)
Comment (imanushin) [inline deployment/SolutionCop.nuspec]: fixed

File: src/SolutionCop.DefaultRules.Tests/NuGet/NuspecHasTheSameVersionsWithPackagesConfigTest.cs (hunk 7 above)
Comment (imanushin) [inline src/SolutionCop.DefaultRules.Tests/NuGet/NuspecHasTheSameVersionsWithPackagesConfigTest.cs]: fixed

File: src/SolutionCop.DefaultRules.Tests/NuGet/NuspecHasTheSameVersionsWithPackagesConfigTest.cs (hunk 9 above)
Comment (imanushin) [inline src/SolutionCop.DefaultRules.Tests/NuGet/NuspecHasTheSameVersionsWithPackagesConfigTest.cs]: fixed

File: src/SolutionCop.DefaultRules.Tests/NuGet/NuspecHasTheSameVersionsWithPackagesConfigTest.cs (hunk 8 above)
Comment (imanushin) [inline src/SolutionCop.DefaultRules.Tests/NuGet/NuspecHasTheSameVersionsWithPackagesConfigTest.cs]: fixed

**Comment by imanushin** [comment]
all requirements were applied.
also project/package parsing logic was extracted to separate file (to reuse in future)

**Pull request merged**